samples/dnn/segmentation.cpp

```diff
@@ -146,29 +146,35 @@ int main(int argc, char **argv)
         blobFromImage(frame, blob, scale, Size(inpWidth, inpHeight), mean, swapRB, false);
         //! [Set input blob]
         net.setInput(blob);
-        //! [Make forward pass]
-        Mat score = net.forward();
+        //! [Set input blob]
+
         if (modelName == "u2netp")
         {
-            Mat mask, thresholded_mask, foreground_overlay, background_overlay, foreground_segmented;
-            mask = cv::Mat(score.size[2], score.size[3], CV_32F, score.ptr<float>(0, 0));
-            mask.convertTo(mask, CV_8U, 255);
-            threshold(mask, thresholded_mask, 0, 255, THRESH_BINARY + THRESH_OTSU);
-            resize(thresholded_mask, thresholded_mask, Size(frame.cols, frame.rows), 0, 0, INTER_AREA);
+            vector<Mat> output;
+            net.forward(output, net.getUnconnectedOutLayersNames());
+
+            Mat pred = output[0].reshape(1, output[0].size[2]);
+            pred.convertTo(pred, CV_8U, 255.0);
+            Mat mask;
+            resize(pred, mask, Size(frame.cols, frame.rows), 0, 0, INTER_AREA);
+
             // Create overlays for foreground and background
-            foreground_overlay = Mat::zeros(frame.size(), frame.type());
-            background_overlay = Mat::zeros(frame.size(), frame.type());
+            Mat foreground_overlay = Mat::zeros(frame.size(), frame.type());
+
             // Set foreground (object) to red and background to blue
-            foreground_overlay.setTo(Scalar(0, 0, 255), thresholded_mask);
-            Mat inverted_mask;
-            bitwise_not(thresholded_mask, inverted_mask);
-            background_overlay.setTo(Scalar(255, 0, 0), inverted_mask);
+            std::vector<Mat> channels;
+            split(foreground_overlay, channels);
+            channels[2] = mask;  // Red foreground
+            merge(channels, foreground_overlay);
+
```

Comment:
split is not needed here. You can create single channel Mat with zeros and then use it several times for merge, e.g.:
```
Mat all_zeros = Mat::zeros(frame.size(), CV_8UC1);
vector<Mat> channels = {all_zeros, all_zeros, mask};
merge(channels, foreground_overlay);
```